src/ceph.c

```diff
@@ -522,18 +517,24 @@ static size_t count_parts(char const *key) {
  */
 static int parse_keys(char *buffer, size_t buffer_size, const char *key_str) {
   char tmp[2 * buffer_size];
+  size_t tmp_size = sizeof(tmp);
 
   if (buffer == NULL || buffer_size == 0 || key_str == NULL ||
       strlen(key_str) == 0)
     return EINVAL;
-
-  if ((count_parts(key_str) > 2) && has_suffix(key_str, ".type")) {
-    /* strip ".type" suffix iff the key has more than two parts. */
-    size_t sz = strlen(key_str) - strlen(".type") + 1;
-
-    if (sz > sizeof(tmp))
-      sz = sizeof(tmp);
-    sstrncpy(tmp, key_str, sz);
+  /* Strip suffix if it is ".type" or one of latency metric suffix. */
+  if (count_parts(key_str) > 2) {
```

Comment:
At your option:

```c
/* initialize tmp */
sstrncpy(tmp, key_str, sizeof(tmp));

char const *cut_suffixes[] = {".type", ".avgcount", ".sum", ".avgtime"};
for (size_t i = 0; i < STATIC_ARRAY_SIZE(cut_suffixes); i++) {
  if (has_suffix(key_str, cut_suffixes[i])) {
    cut_suffix(tmp, sizeof(tmp), key_str, cut_suffixes[i]);
    break;
  }
}
```

This has less repetition and makes it easier to see just which suffixes are being cut.